### ISSUE TITLE
aria-hidden-focus violation

### ISSUE THREAD

**Issue by smellai** [floating-ui/floating-ui#2462]
**Describe the bug**
Deque's axe checks report a violation for the guard element:

```
Detected the following accessibility violations!

     1. aria-hidden-focus (ARIA hidden element must not be focusable or contain focusable elements)

        For more info, visit https://dequeuniversity.com/rules/axe/4.6/aria-hidden-focus?application=axeAPI.

        Check these nodes:

        - html: <span data-type="outside" tabindex="0" aria-hidden="true" data-floating-ui-focus-guard="" style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"></span>
          summary: Fix all of the following:
                     Focusable content should have tabindex='-1' or be removed from the DOM
```

**To Reproduce**
Check the span element rendered in [one of the examples](https://codesandbox.io/s/stoic-bas-frzus0?file=/src/App.tsx)

**Expected behavior**
aria-hidden element should not be focusable

**Comment by atomiks** [comment]
See https://github.com/floating-ui/floating-ui/pull/2298#issuecomment-1518101512

**Comment by smellai** [comment]
ok thanks 🙏
